lib/matplotlib/backends/backend_qt.py

```diff
@@ -268,40 +268,47 @@ def mouseEventCoords(self, pos=None):
         return x * self.device_pixel_ratio, y * self.device_pixel_ratio
 
     def enterEvent(self, event):
+        qt_mods = QtWidgets.QApplication.instance().queryKeyboardModifiers()
```

Comment:
```suggestion
```

Reply:
This does not seem to be used?

Reply:
Good catch, this was something about qt caching for enterEvent (so that figure_enter_event reports the correct modifiers) that I had forgotten to fully implement, now fixed.  Other backends seem OK in this respect.

Reply:
attn @tacaswell in case you want to look at this specific point again.